modules/core/src/test/java/org/apache/ignite/internal/processors/service/GridServiceProcessorAbstractSelfTest.java

```diff
@@ -293,20 +292,15 @@ public void testGetServicesByName() throws Exception {
 
         g.services().deployMultiple(name, new DummyService(), nodeCount() * 2, 3);
 
-        GridTestUtils.retryAssert(log, 50, 200, new CA() {
-            @Override public void apply() {
-                int cnt = 0;
+        AtomicInteger cnt = new AtomicInteger(0);
 
-                for (int i = 0; i < nodeCount(); i++) {
-                    Collection<DummyService> svcs = grid(i).services().services(name);
+        for (int i = 0; i < nodeCount(); i++) {
+            Collection<DummyService> svcs = grid(i).services().services(name);
 
-                    if (svcs != null)
-                        cnt += svcs.size();
-                }
-
-                assertEquals(nodeCount() * 2, cnt);
-            }
-        });
+            if (svcs != null)
+                cnt.addAndGet(svcs.size());
+        }
+        assertTrue(GridTestUtils.waitForCondition(() -> cnt.get() == nodeCount() * 2, 10000));
```

Comment:
Semantic is broken here. 'cnt' is calculated only once and will never be changed. 
So, if condition doesn't met there is no sense to wait neither 10 sec nor more nor less.